### PR TITLE
Create workflow unit test pattern

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

The purpose of this PR was simply to create a pattern for unit testing code in workflows.  It extracted some (fairly simple) code in the nightly product functional testing, created standalone files with that code, called that code from workflow, created unit tests for those files, and then created a new workflow that runs the unit tests.  As this was successful, this is the pattern that will be followed for unit testing all workflow code; in particular, it will be used to improve the quality of the deployment code.

## 💭 Motivation and context ##

Closes:  #1164

## 🧪 Testing ##

- Created code, tests, and workflows
- Ran unit tests by hand and by workflow and by CI/CD pipeline

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*

Mostly.  I did delete one random Markdown file that was incorporated into Nanda's testing README but hadn't been deleted.

- [x] Changes are sized such that they do not touch excessive number of files.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Review by adhilto — COMMENTED** [review]
Overall looks good, just one small recommendation.

File: .github/workflows/test.yaml
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/test.yaml:1]: Hah!  Good catch!  That is completely unrelated.  It was a test workflow for when I was trying to figure out why Act doesn't support the needs keyword.  I will delete this file.

[the remark this answers is not in the document]
Comment (adhilto) [reply .github/workflows/test.yaml:1]: I thought there was a chance it was something like that! Glad I asked.

[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/test.yaml:1]: Done.

**Review by adhilto — APPROVED** [review]

**Comment by james-garriss** [comment]
@nanda-katikaneni , this one is ready.

**Pull request merged**